test/system/src/test/java/io/pravega/test/system/ReadWriteAndAutoScaleWithFailoverTest.java

```diff
@@ -0,0 +1,468 @@
+        Thread.sleep(ZK_DEFAULT_SESSION_TIMEOUT);
+        log.info("Scaling down controller instances from 3 to 2");
+
+        currentWriteCount2 = testState.eventWriteCount.get();
+        currentReadCount2 = testState.eventReadCount.get();
+        log.info("Read count: {}, write count: {} after controller failover after sleep", currentReadCount2, currentWriteCount2);
+        //ensure writes are happening
+        assertTrue(currentWriteCount2 > currentWriteCount1);
+        //ensure reads are happening
+        assertTrue(currentReadCount2 > currentReadCount1);
+
+        //Scale down SSS, controller to 1 instance each.
+        segmentStoreInstance.scaleService(1, true);
+        controllerInstance.scaleService(1, true);
+        //zookeeper will take about 60 seconds to detect that the node has gone down
+        Thread.sleep(new Random().nextInt(50000) + ZK_DEFAULT_SESSION_TIMEOUT);
+        log.info("Scaling down  to 1 controller, 1 SSS instance");
+
+        currentWriteCount1 = testState.eventWriteCount.get();
+        currentReadCount1 = testState.eventReadCount.get();
+        log.info("Stop write flag status: {}, stop read flag status: {} ", testState.stopWriteFlag.get(), testState.stopReadFlag.get());
+        log.info("Read count: {}, write count: {} with SSS and controller failover after sleep", currentReadCount1, currentWriteCount1);
+    }
+
+
+
+    private CompletableFuture<Void> startWriting(final EventStreamWriter<Long> writer) {
+        return CompletableFuture.runAsync(() -> {
+            while (!testState.stopWriteFlag.get()) {
+                try {
+                    long value = testState.eventData.incrementAndGet();
+                    Thread.sleep(100);
+                    log.debug("Event write count before write call {}", value);
+                    writer.writeEvent(String.valueOf(value), value);
+                    log.debug("Event write count before flush {}", value);
+                    writer.flush();
+                    testState.eventWriteCount.getAndIncrement();
+                    log.debug("Writing event {}", value);
+                } catch (InterruptedException e ) {
+                        log.error("Error in sleep: ", e);
+                        break;
+                } catch (ConnectionClosedException e) {
+                        log.warn("Test exception in writing events: ", e);
+                        continue;
+                } catch (Throwable e) {
+                    log.error("Test exception in writing events: ", e);
+                        break;
+                }
+            }
+        }, executorService);
+    }
+
+    private CompletableFuture<Void> startReading(final EventStreamReader<Long> reader) {
+        return CompletableFuture.runAsync(() -> {
+            log.info("Exit flag status {}", testState.stopReadFlag.get());
+            log.info("Read count {} and write count {}", testState.eventReadCount.get(), testState.eventWriteCount.get());
```

Comment:
I'm not sure why you have two consecutive log info lines, please merge them and have a single line.